cranelift/filetests/filetests/isa/riscv64/select_spectre_guard.clif

```diff
@@ -0,0 +1,1089 @@
+test compile precise-output
+set unwind_info=false
+target riscv64
+
+function %f(i8, i8, i8) -> i8 {
+block0(v0: i8, v1: i8, v2: i8):
+  v3 = iconst.i8 42
+  v4 = icmp eq v0, v3
+  v5 = select_spectre_guard.i8 v4, v1, v2
+  return v5
+}
+
+; VCode:
+; block0:
+;   li t1,42
+;   andi t4,a0,255
+;   andi t1,t1,255
```

Comment:
It would be nice if `normalize_cmp_value` didn't normalize constants whose value already fit in the target type, but there's no need to fix that on this PR.